Postal/IMAPSession.swift

```diff
@@ -25,37 +25,42 @@
 import Swift
 import libetpan
 
-private typealias Progress = @convention(c) (Int, Int, UnsafeMutablePointer<Void>) -> Void
+private typealias Progress = @convention(c) (Int, Int, UnsafeMutableRawPointer?) -> Void
 
-public typealias ProgressHandler = (current: Int, maximum: Int) -> ()
+public typealias ProgressHandler = (_ current: Int, _ maximum: Int) -> ()
 
 final class IMAPSession {
     let configuration: Configuration
     let imap: UnsafeMutablePointer<mailimap>
     
-    private(set) var capabilities: IMAPCapability = []
-    private(set) var defaultNamespace: IMAPNamespace? = nil
-    private var serverIdentity = IMAPIdentity([:])
+    fileprivate(set) var capabilities: IMAPCapability = []
+    fileprivate(set) var defaultNamespace: IMAPNamespace? = nil
+    fileprivate var serverIdentity = IMAPIdentity([:])
     
-    private var selectedFolder: String = ""
+    fileprivate var selectedFolder: String = ""
     
     var logger: Logger? {
         didSet {
             if logger != nil {
-                mailimap_set_logger(imap, _logger, UnsafeMutablePointer(Unmanaged.passRetained(self).toOpaque()))
+                mailimap_set_logger(imap, _logger, Unmanaged.passRetained(self).toOpaque())
             } else {
                 mailimap_set_logger(imap, nil, nil)
             }
         }
     }
     
-    private let _logger: IMAPLogger = { (_: UnsafeMutablePointer<mailimap>, logType: Int32, buffer: UnsafePointer<Int8>, size: Int, context: UnsafeMutablePointer<Void>) in
-        guard let logType = ConnectionLogType(rawType: logType) else { return }
-        guard size > 0 else { return }
+    fileprivate let _logger: IMAPLogger = { (_: UnsafeMutablePointer<mailimap>?, logType: Int32, buffer: UnsafePointer<CChar>?, size: Int, context: UnsafeMutableRawPointer?) in
+        guard
+            size > 0,
+            let context = context,
+            let logType = ConnectionLogType(rawType: logType),
+            let buffer = buffer
+        else { return }
 
-        let session = Unmanaged<IMAPSession>.fromOpaque(COpaquePointer(context)).takeUnretainedValue()//.takeRetainedValue()
-
-        if let str = String(data: NSData(bytes: buffer, length: size - 1), encoding: NSUTF8StringEncoding) where !str.isEmpty {
+        let session = Unmanaged<IMAPSession>.fromOpaque(context).takeRetainedValue()  //.takeUnretainedValue()
+        
+        //if let str = String(data: Data(bytes: UnsafePointer<UInt8>(buffer), count: size - 1), encoding: String.Encoding.utf8) , !str.isEmpty {
```

Comment:
-> I think we were using the size to avoid overflows (non zero ending string). We may use a parameter to String.fromUTF8CString providing an optional maximum size)